### PR TITLE
Display unit price in detailed cart product list

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Display product unit price . |
| Type? | improvement |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-722 |
| How to test? | Create a product with unit price and add it to your cart |

cart_-_presta-dev

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
Wow so clean ! Nice contribution Julien :+1:

File: src/Adapter/Cart/CartPresenter.php [hunk 1]
```diff
@@ -91,11 +91,17 @@ private function presentProduct(array $rawProduct)
...
         if ($this->includeTaxes()) {
+            $rawProduct['price_amount'] = $rawProduct['price_wt'];
             $rawProduct['price'] = $this->priceFormatter->format($rawProduct['price_wt']);
         } else {
+            $rawProduct['price_amount'] = $rawProduct['price'];
             $rawProduct['price'] = $rawProduct['price_tax_exc'] = $this->priceFormatter->format($rawProduct['price']);
         }
 
+        if ($rawProduct['price_amount'] && $rawProduct['unit_price_ratio'] > 0) {
+            $rawProduct['unit_price'] = $rawProduct['price_amount'] / $rawProduct['unit_price_ratio'];
```
Comment (mickaelandrieu) [inline src/Adapter/Cart/CartPresenter.php]: only 1 question: what happens if you have a float and an integer ? Should we not control the precision of this result ?

**Comment by Quetzacoalt91** [comment]
Nice screenshot.

File: src/Adapter/Cart/CartPresenter.php (hunk 1 above)
Comment (julienbourdeau) [inline src/Adapter/Cart/CartPresenter.php]: We keep the largest precision here, then the ProductPresenter will format this value properly.

**Comment by mickaelandrieu** [comment]
Thank you @julienbourdeau

**Pull request merged**